### ISSUE TITLE
feat(beam): wire BeamIndirectionTable into beam-search expansion (post-#12407)

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12964]
## Parent context

Follow-on to #12409 (`BeamIndirectionTable` slot table over `BatchSession.StepBatch`), which shipped the dependency-free slice to trunk. Depends on #12407 (`feat(beam): zero-allocation beam search engine spine`), still **OPEN** — `internal/beam/beam.go`, `pool.go`, `arena.go` do not exist on any ref.

## What shipped in #12409

`internal/beam/indirection.go` (reference-counted `BeamIndirectionTable`, zero-copy Fork, O(B*MaxSteps) Reorder, 0 allocs/op) and `internal/beam/batch_driver.go` (`BatchDriver.Step` -> real `model.BatchSession.StepBatch` / `StepBatchActive`, `LastStepMACs` receipt).

## Problem

The table + driver are standalone: nothing wires the indirection table into actual beam-search hypothesis expansion, because the beam engine #12407 owns (`BeamNode` arena, `BeamPool`, survivor selection) does not exist yet. #12409's done condition (decoding B=4 beams using StepBatch and BeamIndirectionTable across branch forks) is therefore only partially achievable today.

## Intended outcome (after #12407 lands)

1. Wire `BeamIndirectionTable` into #12407's beam expansion: on survivor re-selection at step t, `Fork(parent, child, t)` inherits the parent prefix slot pointers and assigns the child a fresh physical slot.
2. Map each surviving beam's logical position to a physical KV slot consumed by the model's batched decode so branch forks need zero float32 KV copies.
3. Add an end-to-end witness decoding B=4 beams with single weight-stream bandwidth (one `StepBatch` per step) and `0 allocs/op` across forks, asserting bit-identical logits vs serial execution.

## Start blocked by

#12407 (the beam engine spine must land first).

## Witness

```bash
go test -v ./internal/beam -run TestBeamZeroAllocKV
go test -bench=BenchmarkBeamIndirectionStep -benchmem ./internal/beam
go test ./internal/beam/... -count=1
```
